walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,50 @@
 
 package com.google.sps;
 
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
 
+/**
+ * Class that finds suitable times for a group of people to meet.
+ **/
 public final class FindMeetingQuery {
+  /**
+   * Returns a list of TimeRanges where a group of people can meet,
+   * restrained by the events they have throughout the day.
+   *
+   * Part the algorithm iterates through the list of events,
+   * finding potential time of meetings in-between events. The second
+   * part then goes through these potential meeting times and looks for
+   * those that satisfy the amount of time that needs to be allotted, as
+   * specified in the request.
+   *
+   * @param {Collection<Event>} events
+   * @param {MeetingRequest} request
+   */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    List<Event> eventsList = new ArrayList<>(events);
+    Collection<TimeRange> timeOfMeetings = new ArrayList<>();
+    int potentialMeetingTimeStart = TimeRange.START_OF_DAY;
+    eventsList.removeIf(event -> !request.getAttendees().containsAll(event.getAttendees()));
+    Collections.sort(eventsList, (Event e1, Event e2) ->
+        TimeRange.ORDER_BY_START.compare(e1.getWhen(), e2.getWhen()));
+    for(Event conflictingEvent : eventsList) {
+      int conflictingEventStart = conflictingEvent.getWhen().start();
+      int conflictingEventEnd = conflictingEvent.getWhen().end();
+      if(conflictingEventStart > potentialMeetingTimeStart) {
+        timeOfMeetings.add(TimeRange.fromStartEnd(potentialMeetingTimeStart,
+            conflictingEventStart, /*inclusiveEndBound=*/false));
+      }
+      if(conflictingEventEnd > potentialMeetingTimeStart) {
+        potentialMeetingTimeStart = conflictingEventEnd;
+      }
```

Comment:
I think you probably don't need this extra conditional and can include this line in the block above. It *should* always be true that:
if conflictingEventStart > potentialMeetingTimeStart 
then
conflictingEventEnd > potentialMeetingTimeStart

Reply:
|--A--|
     |--B--|
If we have a case like this, where person B has an event that starts some minutes after A's, the start of event B isn't greater than the potential meeting start, so a meeting would be scheduled from the potential meeting start (in this case the end of A's event) to the end of the day, as opposed to from the end of B's event (as it should be with the conditional).

Reply:
Ah, yup, you are right! Ignore me.